internal/cli/atlas/deployments/setup.go

```diff
@@ -139,7 +177,74 @@ Creating your cluster %s [this might take several minutes]
 	if _, err := opts.podmanClient.CreateNetwork(ctx, opts.LocalNetworkName()); err != nil {
 		return err
 	}
+	return nil
+}
+
+func (opts *SetupOpts) planSteps(ctx context.Context) (steps int, needPodmanSetup bool, needToPullImages bool) {
+	steps = 2
+	needPodmanSetup = false
+	needToPullImages = false
+
+	setupState := opts.podmanClient.Diagnostics(ctx)
 
+	if !setupState.MachineFound || setupState.MachineState != "running" {
+		steps++
+		needPodmanSetup = true
+	}
+
+	foundMongod := false
+	foundMongot := false
+	for _, image := range setupState.Images {
+		foundMongod = foundMongod || image == opts.MongodDockerImageName()
+		foundMongot = foundMongot || image == options.MongotDockerImageName
+	}
+
+	if !foundMongod || !foundMongot {
+		steps++
+		needToPullImages = true
+	}
+	return steps, needPodmanSetup, needToPullImages
+}
+
+func (opts *SetupOpts) createLocalDeployment(ctx context.Context) error {
+	if err := podman.Installed(); err != nil {
+		return err
+	}
+
+	steps, needPodmanSetup, needToPullImages := opts.planSteps(ctx)
+	currentStep := 1
+	longWaitWarning := ""
```

Comment:
I think for future work it would be good to investigate using a state machine lib